### PR TITLE
Hw 26 part 1

### PR DESCRIPTION
**Pull request by ilya449**

### CODE REVIEW

File: src/main/java/com/internet/shop/controller/RegistrationController.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String name = req.getParameter("name");
+        String login = req.getParameter("login");
+        String pass = req.getParameter("pass");
+        String repeatedPass = req.getParameter("repeatedPass");
+        if (pass.equals(repeatedPass)) {
```
Comment (DariaPikul) [inline src/main/java/com/internet/shop/controller/RegistrationController.java:32]: What if we receive the empty Strings?

File: src/main/java/com/internet/shop/controller/AddProductController.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/view/product/addition.jsp")
+                .forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String name = req.getParameter("name");
```
Comment (DariaPikul) [inline src/main/java/com/internet/shop/controller/AddProductController.java:29]: Same check

File: src/main/java/com/internet/shop/controller/RegistrationController.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            User newUser = new User(name, login, pass);
+            userService.create(newUser);
+            resp.sendRedirect(req.getContextPath() + "/");
```
Comment (DariaPikul) [inline src/main/java/com/internet/shop/controller/RegistrationController.java:35]: We have to create the shopping carts at the moment of creating our users

**Review by DariaPikul — CHANGES_REQUESTED** [review]
Please, implement the method that allows users to add new products to DB.
We'll use it later for admins.

File: src/main/java/com/internet/shop/controller/RegistrationController.java (hunk 1 above)
Comment (ilya449) [reply src/main/java/com/internet/shop/controller/RegistrationController.java:32]: I've missed it, thank you

File: src/main/java/com/internet/shop/controller/RegistrationController.java (hunk 3 above)
Comment (ilya449) [reply src/main/java/com/internet/shop/controller/RegistrationController.java:35]: > Please, implement the method that allows users to add new products to DB.
> We'll use it later for admins.

I have the AddProductController that allows users to add products. But I forgot to put the link in index.jsp, thank you for your reminding

Comment (DariaPikul) [reply src/main/java/com/internet/shop/controller/RegistrationController.java:42]: Then what about adding the products to the shopping cart?

File: src/main/java/com/internet/shop/controller/AddProductController.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import com.internet.shop.model.Product;
+import com.internet.shop.service.ProductService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/product/addition")
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controller/AddProductController.java:13]:
```suggestion
@WebServlet("/product/add")
```

File: src/main/java/com/internet/shop/controller/InjectDataController.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import com.internet.shop.service.UserService;
+import java.io.IOException;
+import java.util.ArrayList;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/injectData")
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controller/InjectDataController.java:18]: https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_16_1.html

File: src/main/java/com/internet/shop/controller/RegistrationController.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String name = req.getParameter("name");
+        String login = req.getParameter("login");
+        String pass = req.getParameter("pass");
+        String repeatedPass = req.getParameter("repeatedPass");
+        if (name.length() > 0 && login.length() > 0
+                && pass.length() > 0 && pass.equals(repeatedPass)) {
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controller/RegistrationController.java:38]: we can use html tags to check is the field empty
find the title: Input Restrictions
https://www.w3schools.com/html/html_form_input_types.asp

File: src/main/java/com/internet/shop/controller/AddProductController.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        req.getRequestDispatcher("/WEB-INF/view/product/addition.jsp")
+                .forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String name = req.getParameter("name");
+        String price = req.getParameter("price");
+        if (name.length() > 0 && price.length() > 0) {
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controller/AddProductController.java:31]: https://www.w3schools.com/html/html_form_input_types.asp

File: src/main/java/com/internet/shop/controller/RegistrationController.java (hunk 3 above)
Comment (ilya449) [reply src/main/java/com/internet/shop/controller/RegistrationController.java:42]: BuyProductController adds products to the shopping cart

File: src/main/java/com/internet/shop/controller/RegistrationController.java (hunk 6 above)
Comment (ilya449) [reply src/main/java/com/internet/shop/controller/RegistrationController.java:38]: Added html tags, and Bohdan allowed not to remove this simple check

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**